### PR TITLE
[tune] fix restore error at tune.run() 

### PR DESCRIPTION
**Pull request by pengzhenghao**
Sorry for the messed up PR yesterday... I am the first-time contributor and not familiar with how git runs... This PR cleans up the irrelevant commits.

## What do these changes do?

Fix an error of restoring checkpoint at `tune.run()`.

Add a small black box test for restoring.

## Related issue number

The same as closed PR #4715 

Solve an error brought by #4490 

Closes issue #4714 

## Linter

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by richardliaw** [comment]
@PengZhenghao can you address https://travis-ci.com/ray-project/ray/jobs/197103865

Thanks!

File: python/ray/tune/tests/test_tune_restore.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import unittest
+
+import ray
+from ray import tune
+from ray.rllib import _register_all
+
+
+class TuneRestoreTest(unittest.TestCase):
+    def setUp(self):
+        self.curdir = os.path.abspath('.')
```
Comment (richardliaw) [inline python/ray/tune/tests/test_tune_restore.py]:
```suggestion
        self.curdir = os.path.abspath(".")
```

This should address https://travis-ci.com/ray-project/ray/jobs/197103865

**Review by richardliaw — APPROVED** [review]

File: python/ray/tune/tests/test_tune_restore.py (hunk 1 above)
Comment (PengZhenghao) [reply python/ray/tune/tests/test_tune_restore.py]: That's great! Could you give me some reference on why single quotes causes error in travis-ci? What's more, are there other mistakes which are easily made, like this one?

**Comment by richardliaw** [comment]
Oh it’s because of our style requirements.

Be sure to run ‘scripts/format.sh’ (found in the top of the ray directory)
to identify any formatting/style errors!

File: python/ray/tune/tune.py [hunk 2]
```diff
@@ -307,15 +309,16 @@ def run_experiments(experiments,
...
+            search_alg=search_alg,
+            scheduler=scheduler,
+            with_server=with_server,
+            server_port=server_port,
+            verbose=verbose,
+            resume=resume,
+            queue_trials=queue_trials,
+            reuse_actors=reuse_actors,
+            trial_executor=trial_executor,
+            raise_on_failed_trial=raise_on_failed_trial)
```
Comment (pengzhenghao) [inline python/ray/tune/tune.py]: It's wired that after I running `scripts/format.sh` at my local on the latest commit, it seems that all your modifications in this commit are reversed and turn to my latest "formatting" commit.

I am using macOS Mojave with python 3.6.0, latest Ray.

**Comment by richardliaw** [comment]
Oh make sure yapf==0.23;

We should put that in the docs somewhere - thanks for pointing this out!

**Comment by ZhongxiaYan** [comment]
I also have the same issue with restore. I can pass in restore=<path to checkpoint file>, but the state does not restore. In particular I'm trying to restore checkpoints from previous tune.run on rllib experiments (like PPO, DQN, etc), but the training always starts from scratch instead of from the checkpoint. Any suggestions?

**Pull request merged**